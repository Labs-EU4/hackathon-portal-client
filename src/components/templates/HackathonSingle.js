import React, { useState } from "react";
import styled from "styled-components";
import { useParams, useHistory } from "react-router-dom";
import { useSelector, useDispatch } from "react-redux";

import HackathonProjectsPage from '../views/HackathonProjectsPage';
import AddTeammates from '../templates/AddTeammates';
import { media } from "../../assets/styles/variables/media";
import { H2, H3 } from "../atoms/Heading";
import { BoldSpan, NormalSpan } from "../atoms/Span";
import { CardWide } from "../atoms/Card";
import { IconLetter } from "../atoms/IconLetter";
import { Paragraph } from "../atoms/Paragraph";
import Button from "../atoms/Button";
import userImg from "../../assets/images/user_icon.svg";
import eventImg from "../../assets/images/event-img.jpg";
// import { useEventTeam } from "../../hooks";

import {
  registerEvent,
  unregisterEvent
} from "../../store/eventParticipants/actions";

import { useParticipants, useEventTeam, useTeams, useEvent } from "../../hooks";
import Spinner from "../molecules/Spinner";

export const SpanContent = styled(BoldSpan)`
  ${props => props.theme.fontSize.medium};
  font-weight: bold; 
  text-transform: capitalize;
`;

export const PTags = styled(Paragraph)`
  display: inline-block;
  text-align: center;
  border: 1px solid #e9b75f;
  border-radius: 6px;
  color: #212121;
  font-weight: bold;
  font-size: 14px;
  text-transform: uppercase;
  margin: 5px 5px 5px 0;
  padding: 7px 8px;
`;

const TitleContainer = styled.div`
  display: flex;
  align-items: baseline;
  justify-content: center;
`;

const ButtonsDashGroup = styled.div`
  a,
  button {
    display: block;
    margin: 0 0 10px 0;
  }
`;

const TagsGroup = styled.div`
  margin-bottom: 20px;
`;

const HackathonSingle = ({ eventId, setEventId, isEventModalOpen, setIsEventModalOpen }) => {
  // const { id } = useParams();
  const id = eventId;
  const [ isAddJudgeOpen, setIsAddJudgeOpen ] = useState(false);
  const [ isSubmissionsPageOpen, setIsSubmissionsPageOpen ] = useState(false);
  const [ isSlideForm, setIsSlideForm ] = useState(false);
  const history = useHistory();
  const dispatch = useDispatch();
  const { userId } = useSelector(state => state.currentUser);
  const [participants, fetchParticipants] = useParticipants(id);
  const [team] = useEventTeam(id);
  const [teams] = useTeams(id);
  const createdTeam = teams.find(t => t.team_lead === userId);

  const [data, loading] = useEvent(id, isAddJudgeOpen);

  // Filter out event by URL param & grab user ID
  const [
    {
      creator_id,
      event_title,
      event_description: description,
      start_date,
      end_date,
      guidelines,
      participation_type,
      tag_name,
      location,
      organizer_email,
      organizer_name,
      organizer_profile_pic,
      rubrics
    }
  ] = data?.body || [
    {
      creator_id: 0,
      event_title: "",
      event_description: "",
      start_date: null,
      end_date: null,
      guidelines: "",
      participation_type: "",
      tag_name: [],
      location: "",
      organizer_email: "",
      organizer_name: "",
      organizer_profile_pic: [],
      rubrics: []
    }
  ];

  // Date formatting
  const formattedStartDate = new Date(start_date).toLocaleDateString();
  const formattedEndDate = new Date(end_date).toLocaleDateString();

  // Event is open or closed for registration
  const userCallback = p => p.user_id === userId;
  const today = new Date().getTime();
  const startTime = new Date(start_date).getTime();
  const endTime = new Date(end_date).getTime();
  const isOpen = today <= startTime;
  const isTeamLead = createdTeam;
  const isRegistered = participants.find(userCallback) || isTeamLead;
  const isEventCreator = creator_id === userId;
  const isTeamMember = team.find(userCallback) || isEventCreator;
  const isEnded = today > endTime;
  const individualParticipation = participation_type === "individual";

  // Number of participants registered
  const registeredPartcipants = participants.length;

  // Redacting user emails before rendering
  let redactedEmail = organizer_email.split("");
  let atIndex = redactedEmail.indexOf("@");
  let emailUser = redactedEmail.slice(0, atIndex);
  let emailHost = redactedEmail.slice(atIndex, redactedEmail.length);

  emailUser = emailUser
    .map((redact, index) => {
      if (index === 0 || index === emailUser.length - 1) {
        return redact;
      } else {
        redact = "*";
        return redact;
      }
    })
    .concat(emailHost)
    .join("");

  // Grab the first letter of title
  const initial = event_title[0] || "U";

  const handleRegistration = e => {
    e.preventDefault();

    if (isRegistered) {
      dispatch(unregisterEvent(id, history));
    } else {
      dispatch(registerEvent(id, history));
    }
    return fetchParticipants();
  };

  const toTittleCase = item => {
    return item
      .split("_")
      .map(word => word.charAt(0).toUpperCase() + word.slice(1))
      .join(" ");
  };

  const handleExit = () => {
    setIsEventModalOpen(false);
    setIsSlideForm(false);
  };

  const renderSingleEvent = () => {
    return (
      <ModalBody>
        {loading ? (
          <Spinner />
        ) : (
          <>
              <StyledEventCard active={isSlideForm}>
                <EventCardLeftColumn>
                  <TitleContainer>
                    <IconLetter>{initial}</IconLetter>
                    <H2>{event_title}</H2>
                  </TitleContainer>
                  <EventImageContainer>
                    {/* Here it will go image of the event */}
                    <EventImg src={eventImg} alt="event-image" /> 
                  </EventImageContainer>
                  <ContentTitle text="Judges" {...{isSlideForm}} />
                  <JudgesContainer>
                    {team.length === 0 ? (
                        <Paragraph>
                          No Judges have been selected for this event
                        </Paragraph>
                      ) : (
                        team.map(member =>
                          <JudgeCard key={member.user_id}>
                            {
                              member.image_url === null ? (
                                <JudgeImg
                                  style={{
                                    width: "80px",
                                    objectFit: "cover",
                                    display: "inline-block"
                                  }}
                                  alt="team member profile pic"
                                  src={userImg}
                                />
                              ) : (
                                member.image_url.map((mem, index) => {
                                  let memberProfile;
                                  memberProfile = JSON.parse(mem);
                                  return (
                                    <JudgeImg
                                      key={index}
                                      style={{
                                        width: "80px",
                                        objectFit: "cover",
                                        borderRadius: "50%"
                                      }}
                                      alt="team member profile pic"
                                      src={memberProfile.avatar}
                                    />
                                  );
                                })
                              )
                            }
                            <JudgeInfo>
                              <p>{member.fullname}</p>
                            </JudgeInfo>
                          </JudgeCard> 
                        )
                      )
                    }
                  </JudgesContainer>
                  <ContentTitle text="About this event" {...{isSlideForm}} />
                  <Paragraph>
                    {description}
                  </Paragraph>
                  <ContentTitle text="Guidelines" {...{isSlideForm}} />
                  <Paragraph>
                    {guidelines}
                  </Paragraph>
                  <ContentTitle text="Rubrics" {...{isSlideForm}} />
                  <TagsGroup>
                    {rubrics.map(rubric => {
                      return <PTags key={rubric}>{toTittleCase(rubric)}</PTags>;
                    })}
                  </TagsGroup>
                  <ContentTitle text="Event Tags" {...{isSlideForm}} />
                  <TagsGroup>
                    {tag_name && tag_name.length !== 0 ? (
                      tag_name.map((tagged, index) => {
                        return <PTags key={index}>{tagged}</PTags>;
                      })
                    ) : (
                      <Paragraph>No tags provided for this event</Paragraph>
                    )}
                  </TagsGroup>
                </EventCardLeftColumn>
                <TagsCardWide active={isSlideForm}>
                  <TagCard>
                    <BoldSpan>Hosted by:</BoldSpan>
                    <UserContainer>
                      {organizer_profile_pic === null ? (
                        <Image src={userImg} alt="user_icon" />
                      ) : (
                        organizer_profile_pic.map((mem, index) => {
                          let memberProfile;
                          memberProfile = JSON.parse(mem);
                          return (
                            <Image
                              key={index}
                              src={memberProfile.avatar}
                              alt="user_icon"
                            />
                          );
                        })
                      )}
                      <PHosted>{organizer_name || emailUser}</PHosted>
                    </UserContainer>
                  </TagCard>
                  <TagCard>
                    <NormalSpan>
                      Location: 
                      <SpanContent>&nbsp;{location}</SpanContent>
                    </NormalSpan>
                    <NormalSpan>
                      Status:
                      <SpanContent>{isOpen ? " Open" : " Closed"}</SpanContent>
                    </NormalSpan>
                    <NormalSpan>
                      Participation type:{" "}
                      <SpanContent>{participation_type}</SpanContent>
                    </NormalSpan>
                    <NormalSpan style={{ marginBottom: "15px" }}>
                      Participants:{" "}
                      <SpanContent>{registeredPartcipants}</SpanContent>
                    </NormalSpan>
                    {!isTeamMember && isOpen ? (
                      <Button
                        color={isRegistered ? "grey" : "green"}
                        {...{
                          anchor: !individualParticipation,
                          onClick: individualParticipation && handleRegistration,
                          to: !individualParticipation && `/dashboard/event/${id}/participant-teams`
                        }}
                      >
                        {isRegistered ? `Unregister` : `Register`}
                      </Button>
                    ) : (
                      !isOpen && (
                        <Button disabled >Registration Closed</Button>
                      )
                    )}
                  </TagCard>
                  {/* <Details>
                    <div>
                      <Paragraph>
                        <BoldSpan>Event starts:</BoldSpan>
                        {formattedStartDate}
                      </Paragraph>
                    </div>

                    <div>
                      <Paragraph>
                        <BoldSpan>Event ends:</BoldSpan>
                        {formattedEndDate}
                      </Paragraph>
                    </div>
                  </Details> */}
                  <ButtonsDashGroup>
                    {isEventCreator && !isEnded && (
                      <Button
                        anchor
                        to={`/dashboard/event/${id}/edit`}
                        onClick={() => setIsEventModalOpen(false)}
                        color="blue"
                      >
                        Edit event
                      </Button>
                    )}
                    {isEventCreator && !isEnded && (
                      <Button
                        size= "wide"
                        color="green"
                        onClick={() => setIsAddJudgeOpen(true)}
                      >
                        Add Judges
                      </Button>
                    )}
                    
                    {isTeamLead && !isEnded && (
                      <Button
                        anchor
                        size= "wide"
                        color="green"
                        to={`/dashboard/event/${id}/participant-teams`}
                      >
                        Add teamate
                      </Button>
                    )}
                    {isRegistered && !isEnded && (
                      <Button
                        anchor
                        size= "wide"
                        color="green"
                        to={`/dashboard/event/${id}/participant_submission`}
                      >
                        Submit Project
                      </Button>
                    )}
                    <Button
                      size= "wide"
                      color="blue"
                      onClick={() => setIsSubmissionsPageOpen(true)}
                    >
                      View submissions
                    </Button>
                  </ButtonsDashGroup>
                </TagsCardWide>
              </StyledEventCard>
          </>
        )}
        <ExitButton 
          onClick={handleExit}
          onMouseOver={() => setIsSlideForm(true)}
          onMouseLeave={() => setIsSlideForm(false)}
          color="primary"
        >X</ExitButton>
      </ModalBody>
    );
  }

  if (isAddJudgeOpen) {
    return (
      <>
        { renderSingleEvent() }
        <AddTeammates 
          {...{id}}
          {...{setEventId}}
          {...{setIsEventModalOpen}} 
          {...{setIsAddJudgeOpen}}
        />
      </>
    );
  }

  if (isSubmissionsPageOpen) {
    return (
      <>
        { renderSingleEvent() }
        <HackathonProjectsPage 
          {...{id}}
          {...{setIsSubmissionsPageOpen}} 
        />
      </>
    );
  }

  if (isEventModalOpen) {
    return renderSingleEvent();
  }
  return null
};

export default HackathonSingle;

const ModalBody = styled.div`
  ${props => props.theme.shadow.box};
  position: absolute; top: 0; left: 0;
  background-color: rgba(0, 0, 0, .4);
  width: 100%; height: 100%;
  padding: 60px 20px;
  z-index: 100;
`;

const StyledEventCard = styled(CardWide)`
  position: relative;
  min-width: calc(100% - 250px); height: calc(100vh - 110px);
  background-color: ${props => props.active ? 'rgba(0, 0, 0, .8)' : props.theme.color.grey.bg};
  transform: ${props => props.active && 'translateY(80%)'};
  padding-left: 45px;
  overflow-y: scroll;
  box-shadow: 3px 3px 10px ${props => props.theme.color.black.regular};
  transition: ${props => props.active && 'transform .5s ease'};

  &::-webkit-scrollbar {
    width: 0; height: 0;
  }
`;

const EventCardLeftColumn = styled.div`
  width: calc(100% - 300px);

  @media ${media.tablet} {
    width: 100%;
  }
`;

const EventImageContainer = styled.figure`
  width: 100%; height: 350px;
  margin-bottom: 10px;
  object-fit: cover;
`;

const EventImg = styled.img`
  width: 100%; height: 100%;
`;

const JudgesContainer = styled.div`
  width: 100%;
  display: flex; flex-wrap: wrap;
  margin-bottom: 10px;
`;

const JudgeCard = styled.div`
  display: flex;
  width: 33%; min-width: 250px; height: 80px;
  margin: 10px 0;
`;

const JudgeImg = styled.img`
  width: 80px;
  object-fit: cover;
  border-radius: 50%;
`;

const JudgeInfo = styled.div`
  ${props => props.theme.flex.column};
  padding: 10px;
`;

export const TagsCardWide = styled(CardWide)`
  ${props => props.theme.shadow.box};
  position: fixed; left: calc(100% - 585px); top: 70px;
  display: ${({ active }) => active && 'none'};
  width: 300px; max-height: calc(100vh - 130px);
  border: 1px solid ${props => props.theme.color.primary.regular};
  padding: 8px 5px;
  overflow-y: scroll;

  @media ${media.tablet} {
    width: 100%;
  }
`;

export const TagCard = styled.div`
  ${props => props.theme.flex.column};
  border: 2px solid ${props => props.theme.color.grey.light};
  margin-bottom: 10px;
  padding: 10px;
`;

export const UserContainer = styled.div`
  ${props => props.theme.flex.columnCenter};
  padding-bottom: 0;
`;

export const Image = styled.img`
  width: 60px;
  height: 60px;
  object-fit: cover;
  border-radius: 30px;
  margin: 10px 0 5px;
`;

export const PHosted = styled(Paragraph)`
  ${props => props.theme.fontSize.medium};
  color: darkgray;
  margin-bottom: 0;
`;

export const ExitButton = styled.p`
  ${props => props.theme.flex.center};
  position: fixed; top: 77px; left: 35px;
  width: 35px; height: 35px;
  background-image: linear-gradient(to right, 
    ${props => props.theme.color.white.regular} 50%, 
    ${props => props.theme.color.primary.regular} 50%, 
    ${props => props.theme.color.primary.regular} 100%);
  background-size: 204%;
  border: 3px solid ${props => props.theme.color.black.regular};
  border-radius: 50%;
  color: ${props => props.theme.color.black.regular};
  transition: all .2s;
  cursor: pointer;

  &:hover {
    transform: scale(1.1);
    background-position: 100%;
    color: ${props => props.theme.color.white.regular};
  }
`;


//NEW COMPONENT
const ContentTitle = ({ text, isSlideForm }) => {
  return (
    <StyledContentTitle active={isSlideForm}>
      <p>{text}</p>
    </StyledContentTitle>
  );
}; 

// export default TitleContent;

const StyledContentTitle = styled.div`
  position: relative;
  width: 100%; height: 20px;
  border-bottom: 2px solid ${props => props.theme.color.grey.border};
  margin-bottom: 20px;

  p {
    display: flex; align-items: baseline;
    position: absolute; top: 100%;
    background-color: ${props => props.active ? 'transparent' : props.theme.color.grey.bg };
    padding: 5px 10px 5px 0;
    font-size: 2rem; font-weight: bold;
    text-transform: uppercase;
    transform: translateY(-45%);
  }
`;