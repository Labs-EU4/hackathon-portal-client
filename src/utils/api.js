import Axios from "axios";
import { toast } from "react-toastify";
import { resetUser } from "../store/user/actions";

const baseURL = process.env.REACT_APP_API_URL;

// if process.env.CONTEXT = "production" {
//   Point to heroku API
// }
// if process.env.BRANCH = "develop" {
  // point to Heroku staging API
// }
// if process.env.CONTEXT = "deploy-preview" {
//   point to deploy-preview HEROKU API (the separate one)
// }



export const axiosWithAuth = token =>
  Axios.create({
    baseURL,
    headers: {
      Authorization: token
    }
  });

export const axios = Axios.create({ baseURL });

export const selectToken = state => state.currentUser.token;

export const showError = message => {
  toast.error(message);
};

export const showSuccess = message => {
  toast.success(message);
};

export function* handleError({ response }, put, history) {
  const { message, statusCode } = response.data;
  if (statusCode === 404) {
    history.push("/not-found");
  }
  if (message === "jwt expired") {
    yield put(resetUser());
  } else showError(`⚠️ ${message}`);
}
