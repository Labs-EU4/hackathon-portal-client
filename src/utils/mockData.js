export const initialState = {
  events: {
    data: [
      {
        id: 1,
        event_title: "EuroHack",
        event_description: "the best event hack hack hack macbook",
        creator_id: 1,
        start_date: "10/09/11",
        end_date: "08/07/13",
        location: "Europe",
        guidelines: "cool hekhkdsjfhsdkjf skdjfhskdjfhkjsdfh skjdfhksjdfh",
        participation_type: "Team",
        category_id: 1,
        requirements: [
          {
            video_url:
              "https://res.cloudinary.com/ibkrice-com/video/upload/v1581259780/Harvesting_ifato8.mp4"
          }
        ],
        ProfileImage: [
          {
            image_url:
              "https://res.cloudinary.com/ibkrice-com/video/upload/v1581259780/Harvesting_ifato8.mp4"
          }
        ],
        rubrics: [
          "Presentation",
          "Product Fit",
          "Functionality",
          "Innovation",
          "Product Design",
          "Extensibility"
        ]
      },
      {
        id: 2,
        event_title: "World",
        event_description: "number one hacker event in the world",
        creator_id: 1,
        start_date: "07/09/11",
        end_date: "08/07/13",
        location: "Asia",
        guidelines: "come well prepared,no cheating,no fighting",
        participation_type: "Team",
        category_id: 2
      }
    ],
    categories: [
      { id: 2, category_name: "Summer Hackton" },
      { id: 3, category_name: "World Hackton" },
      { id: 3, category_name: "Asia Hackton" },
      { id: 3, category_name: "Gamer's Hackton" },
      { id: 3, category_name: "Advanced Hackton" }
    ]
  },

  currentUser: {
    token: 347534574536,
    userId: 1,
    fullname: "Jake Tom",
    email: "8omemail@google.com",
    username: "Jake22",
    bio: "I like coding ",
    image: ""
  },
  submissions: [
    {
      id: 1,
      project_title: "Mock Project",
      participant_or_team_name: "Mock Team",
      event_id: 7,
      video_url: "",
      git_url: "https://www.google.com/",
      project_writeups: "No mock write up",
      submitted_by: 14,
      created_at: "2020-03-02",
      updated_at: "2020-03-02",
      average_rating: 0,
      acted_judges: 0,
      number_of_judges: 0
    }
  ],

  teams: [
    { id: 1, team_name: "The A team", team_lead: 2 },
    { id: 2, team_name: "Hacking central", team_lead: 3 },
    { id: 3, team_name: "blackhats", team_lead: 4 },
    { id: 4, team_name: "Spies", team_lead: 1 }
  ],
  participant_team_members: [
    {
      team_member: 3,
      team_id: 1
    },
    {
      team_member: 4,
      team_id: 1
    },
    {
      team_member: 5,
      team_id: 1
    }
  ]
};
