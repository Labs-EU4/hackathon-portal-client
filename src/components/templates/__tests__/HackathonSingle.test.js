import React from "react";
import { Router } from "react-router";
import { createMemoryHistory } from "history";
import { Provider } from "react-redux";
import { render, cleanup } from "@testing-library/react";
import configureStore from "redux-mock-store";
import "@testing-library/jest-dom/extend-expect";
import HackathonSingle from "../HackathonSingle";
import { initialState } from "../../../utils/mockData";

const history = createMemoryHistory();

afterEach(cleanup);

let jestFeatures;
let mockStore;
let store;

beforeEach(() => {
  mockStore = configureStore();
  // let wrapper;
  store = mockStore(initialState);
  jestFeatures = render(
    <Router history={history}>
      <Provider store={store}>
        <HackathonSingle />
      </Provider>
    </Router>
  );
});

describe("Shows all the text nodes on HackathonSingle.js that are contained on the making an event form", () => {
  //   it("should be displaying the <H3>", () => {
  //     let mainHeader = () => jestFeatures.getByText("Add Teammates");
  //     expect(mainHeader()).toBeInTheDocument();
  //   });
  it("asserts the text copyright text node is rendering", () => {
    expect(
      jestFeatures.getByText("International Crafters Â© 2020")
    ).toBeInTheDocument();
  });
  it("asserts the text alt for img is rendering", () => {
    expect(jestFeatures.getByAltText("LinkedIn")).toBeInTheDocument();
  });
  it("asserts that the component renders properly", () => {
    expect(jestFeatures).toMatchSnapshot();
  });
});
