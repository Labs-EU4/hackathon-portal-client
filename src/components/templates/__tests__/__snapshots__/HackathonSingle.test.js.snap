// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shows all the text nodes on HackathonSingle.js that are contained on the making an event form asserts that the component renders properly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="sc-ifAKCX kjqdgS"
        >
          <div
            class="sc-EHOje cawcRW"
          >
            <a
              href="/dashboard"
            >
              <img
                alt="Hackton - Organise hackathons"
                class="sc-bZQynM cMGtGG"
                src="Hackton-logo.png"
              />
            </a>
            <div
              class="sc-jqCOkK evwRBb"
            >
              <div
                class="sc-htoDjs iAtLjJ"
              >
                <span>
                  <svg>
                    link-burger-icon.svg
                  </svg>
                  Menu
                </span>
                <ul>
                  <li>
                    <a
                      class="sc-dnqmqq gRMNAQ"
                      href="/dashboard"
                    >
                      Dashboard
                    </a>
                  </li>
                  <li>
                    <a
                      class="sc-dnqmqq gRMNAQ"
                      href="/dashboard/profile"
                    >
                      Profile
                    </a>
                  </li>
                </ul>
              </div>
              <div
                class="sc-iwsKbI fnOUWl"
              >
                S
                <ul
                  class="sc-gZMcBi hjGwiG"
                >
                  <li>
                    <a
                      href="/dashboard/profile/edit"
                    >
                      Settings
                    </a>
                  </li>
                  <li>
                    <span>
                      LogOut
                    </span>
                  </li>
                </ul>
              </div>
            </div>
          </div>
        </div>
        <div
          class="sc-cMljjf hEMCcP"
        >
          <div
            class="sc-gzVnrw ehKlJb"
          >
            <a
              class="sc-dnqmqq gRMNAQ"
              href="/dashboard"
            >
              <svg>
                link-dashboard-icon.svg
              </svg>
              Dashboard
            </a>
            <a
              class="sc-dnqmqq gRMNAQ"
              href="/dashboard/profile"
            >
              <svg>
                link-profile-icon.svg
              </svg>
              Profile
            </a>
          </div>
          <div
            class="sc-bwzfXH cwdcdI sc-bRBYWo csJunq"
          >
            <div
              class="sc-gPEVay fSXvAX"
              spacing="center"
            >
              <div
                aria-busy="true"
                class=""
              >
                <svg
                  aria-label="audio-loading"
                  height="100"
                  stroke="#2559a1"
                  viewBox="0 0 44 44"
                  width="100"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <g
                    fill="none"
                    fill-rule="evenodd"
                    stroke-width="2"
                  >
                    <circle
                      cx="22"
                      cy="22"
                      r="1"
                    >
                      <animate
                        attributeName="r"
                        begin="0s"
                        calcMode="spline"
                        dur="1.8s"
                        keySplines="0.165, 0.84, 0.44, 1"
                        keyTimes="0; 1"
                        repeatCount="indefinite"
                        values="1; 20"
                      />
                      <animate
                        attributeName="strokeOpacity"
                        begin="0s"
                        calcMode="spline"
                        dur="1.8s"
                        keySplines="0.3, 0.61, 0.355, 1"
                        keyTimes="0; 1"
                        repeatCount="indefinite"
                        values="1; 0"
                      />
                    </circle>
                    <circle
                      cx="22"
                      cy="22"
                      r="1"
                    >
                      <animate
                        attributeName="r"
                        begin="-0.9s"
                        calcMode="spline"
                        dur="1.8s"
                        keySplines="0.165, 0.84, 0.44, 1"
                        keyTimes="0; 1"
                        repeatCount="indefinite"
                        values="1; 20"
                      />
                      <animate
                        attributeName="strokeOpacity"
                        begin="-0.9s"
                        calcMode="spline"
                        dur="1.8s"
                        keySplines="0.3, 0.61, 0.355, 1"
                        keyTimes="0; 1"
                        repeatCount="indefinite"
                        values="1; 0"
                      />
                    </circle>
                  </g>
                </svg>
              </div>
            </div>
          </div>
        </div>
        <div
          class="sc-gqjmRU gUOFMN"
        >
          <div
            class="sc-VigVT fmYOuc"
          >
            <div>
              <img
                alt="LinkedIn"
                class="sc-jTzLTM dCoQrq"
                src="Icon-linkedin.png"
              />
              <img
                alt="Twitter"
                class="sc-jTzLTM dCoQrq"
                src="Icon-twitter.png"
              />
              <img
                alt="Facebook"
                class="sc-jTzLTM dCoQrq"
                src="Icon-facebook.png"
              />
            </div>
            <div>
              <a
                class="sc-cSHVUG sc-chPdSV tifze"
                href="https://hackton.co/#team"
              >
                Team
              </a>
              <a
                class="sc-cSHVUG sc-chPdSV tifze"
                href="#"
              >
                Privacy
              </a>
              <a
                class="sc-cSHVUG sc-chPdSV tifze"
                href="#"
              >
                Contacts
              </a>
            </div>
            <div>
              <p
                class="sc-kgoBCf sc-kGXeez iYoAqE"
              >
                International Crafters Â© 2020
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="sc-ifAKCX kjqdgS"
      >
        <div
          class="sc-EHOje cawcRW"
        >
          <a
            href="/dashboard"
          >
            <img
              alt="Hackton - Organise hackathons"
              class="sc-bZQynM cMGtGG"
              src="Hackton-logo.png"
            />
          </a>
          <div
            class="sc-jqCOkK evwRBb"
          >
            <div
              class="sc-htoDjs iAtLjJ"
            >
              <span>
                <svg>
                  link-burger-icon.svg
                </svg>
                Menu
              </span>
              <ul>
                <li>
                  <a
                    class="sc-dnqmqq gRMNAQ"
                    href="/dashboard"
                  >
                    Dashboard
                  </a>
                </li>
                <li>
                  <a
                    class="sc-dnqmqq gRMNAQ"
                    href="/dashboard/profile"
                  >
                    Profile
                  </a>
                </li>
              </ul>
            </div>
            <div
              class="sc-iwsKbI fnOUWl"
            >
              S
              <ul
                class="sc-gZMcBi hjGwiG"
              >
                <li>
                  <a
                    href="/dashboard/profile/edit"
                  >
                    Settings
                  </a>
                </li>
                <li>
                  <span>
                    LogOut
                  </span>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
      <div
        class="sc-cMljjf hEMCcP"
      >
        <div
          class="sc-gzVnrw ehKlJb"
        >
          <a
            class="sc-dnqmqq gRMNAQ"
            href="/dashboard"
          >
            <svg>
              link-dashboard-icon.svg
            </svg>
            Dashboard
          </a>
          <a
            class="sc-dnqmqq gRMNAQ"
            href="/dashboard/profile"
          >
            <svg>
              link-profile-icon.svg
            </svg>
            Profile
          </a>
        </div>
        <div
          class="sc-bwzfXH cwdcdI sc-bRBYWo csJunq"
        >
          <div
            class="sc-gPEVay fSXvAX"
            spacing="center"
          >
            <div
              aria-busy="true"
              class=""
            >
              <svg
                aria-label="audio-loading"
                height="100"
                stroke="#2559a1"
                viewBox="0 0 44 44"
                width="100"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g
                  fill="none"
                  fill-rule="evenodd"
                  stroke-width="2"
                >
                  <circle
                    cx="22"
                    cy="22"
                    r="1"
                  >
                    <animate
                      attributeName="r"
                      begin="0s"
                      calcMode="spline"
                      dur="1.8s"
                      keySplines="0.165, 0.84, 0.44, 1"
                      keyTimes="0; 1"
                      repeatCount="indefinite"
                      values="1; 20"
                    />
                    <animate
                      attributeName="strokeOpacity"
                      begin="0s"
                      calcMode="spline"
                      dur="1.8s"
                      keySplines="0.3, 0.61, 0.355, 1"
                      keyTimes="0; 1"
                      repeatCount="indefinite"
                      values="1; 0"
                    />
                  </circle>
                  <circle
                    cx="22"
                    cy="22"
                    r="1"
                  >
                    <animate
                      attributeName="r"
                      begin="-0.9s"
                      calcMode="spline"
                      dur="1.8s"
                      keySplines="0.165, 0.84, 0.44, 1"
                      keyTimes="0; 1"
                      repeatCount="indefinite"
                      values="1; 20"
                    />
                    <animate
                      attributeName="strokeOpacity"
                      begin="-0.9s"
                      calcMode="spline"
                      dur="1.8s"
                      keySplines="0.3, 0.61, 0.355, 1"
                      keyTimes="0; 1"
                      repeatCount="indefinite"
                      values="1; 0"
                    />
                  </circle>
                </g>
              </svg>
            </div>
          </div>
        </div>
      </div>
      <div
        class="sc-gqjmRU gUOFMN"
      >
        <div
          class="sc-VigVT fmYOuc"
        >
          <div>
            <img
              alt="LinkedIn"
              class="sc-jTzLTM dCoQrq"
              src="Icon-linkedin.png"
            />
            <img
              alt="Twitter"
              class="sc-jTzLTM dCoQrq"
              src="Icon-twitter.png"
            />
            <img
              alt="Facebook"
              class="sc-jTzLTM dCoQrq"
              src="Icon-facebook.png"
            />
          </div>
          <div>
            <a
              class="sc-cSHVUG sc-chPdSV tifze"
              href="https://hackton.co/#team"
            >
              Team
            </a>
            <a
              class="sc-cSHVUG sc-chPdSV tifze"
              href="#"
            >
              Privacy
            </a>
            <a
              class="sc-cSHVUG sc-chPdSV tifze"
              href="#"
            >
              Contacts
            </a>
          </div>
          <div>
            <p
              class="sc-kgoBCf sc-kGXeez iYoAqE"
            >
              International Crafters Â© 2020
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
